File: test/aidbox_sdk/snapshots/python/hl7-fhir-r4-core/Composition.py
from typing import Optional, List
from pydantic import *
from ..base.Period import Period
from ..base.CodeableConcept import CodeableConcept
from ..base.DomainResource import DomainResource
from ..base.Reference import Reference
from ..base.Identifier import Identifier
from ..base.BackboneElement import BackboneElement
from ..base.Narrative import Narrative

class Composition_Section(BackboneElement):
    ordered_by: Optional[CodeableConcept] = None
    section: Optional[List[Reference]] = None
    mode: Optional[str] = None
    title: Optional[str] = None
    empty_reason: Optional[CodeableConcept] = None
    author: Optional[List[Reference]] = None
    code: Optional[CodeableConcept] = None
    focus: Optional[Reference] = None
    entry: Optional[List[Reference]] = None
    text: Optional[Narrative] = None

class Composition_Attester(BackboneElement):
    mode: str
    time: Optional[str] = None
    party: Optional[Reference] = None

class Composition_Event(BackboneElement):
    code: Optional[List[CodeableConcept]] = None
    detail: Optional[List[Reference]] = None
    period: Optional[Period] = None

class Composition_RelatesTo(BackboneElement):
    code: str
    target_reference: Optional[Reference] = None
    target_identifier: Optional[Identifier] = None

class Composition(DomainResource):
    category: Optional[List[CodeableConcept]] = None
    date: str
    encounter: Optional[Reference] = None
    section: Optional[List[Composition_Section]] = None
    attester: Optional[List[Composition_Attester]] = None
    type: CodeableConcept
    title: str
    author: list[Reference] = []
    event: Optional[List[Composition_Event]] = None
    custodian: Optional[Reference] = None
    status: str
    identifier: Optional[Identifier] = None
    relates_to: Optional[List[Composition_RelatesTo]] = None
    subject: Optional[Reference] = None
    confidentiality: Optional[str] = None

File: test/aidbox_sdk/snapshots/dotnet/hl7-fhir-r4-core/Composition.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.R4.Core;

public class Composition : DomainResource, IResource
{
    public Base.CodeableConcept[]? Category { get; set; }
    public required string Date { get; set; }
    public Base.ResourceReference? Encounter { get; set; }
    public Composition_Section[]? Section { get; set; }
    public Composition_Attester[]? Attester { get; set; }
    public required Base.CodeableConcept Type { get; set; }
    public required string Title { get; set; }
    public required Base.ResourceReference[] Author { get; set; }
    public Composition_Event[]? Event { get; set; }
    public Base.ResourceReference? Custodian { get; set; }
    public required string Status { get; set; }
    public Base.Identifier? Identifier { get; set; }
    public Composition_RelatesTo[]? RelatesTo { get; set; }
    public Base.ResourceReference? Subject { get; set; }
    public string? Confidentiality { get; set; }

    public class Composition_Section_Section : BackboneElement
    {
        public Base.CodeableConcept? OrderedBy { get; set; }
        public string[]? Section { get; set; }
        public string? Mode { get; set; }
        public string? Title { get; set; }
        public Base.CodeableConcept? EmptyReason { get; set; }
        public Base.ResourceReference[]? Author { get; set; }
        public Base.CodeableConcept? Code { get; set; }
        public Base.ResourceReference? Focus { get; set; }
        public Base.ResourceReference[]? Entry { get; set; }
        public Base.Narrative? Text { get; set; }
    }

    public class Composition_Section : BackboneElement
    {
        public Base.CodeableConcept? OrderedBy { get; set; }
        public Composition_Section_Section[]? Section { get; set; }
        public string? Mode { get; set; }
        public string? Title { get; set; }
        public Base.CodeableConcept? EmptyReason { get; set; }
        public Base.ResourceReference[]? Author { get; set; }
        public Base.CodeableConcept? Code { get; set; }
        public Base.ResourceReference? Focus { get; set; }
        public Base.ResourceReference[]? Entry { get; set; }
        public Base.Narrative? Text { get; set; }
    }

    public class Composition_Attester : BackboneElement
    {
        public required string Mode { get; set; }
        public string? Time { get; set; }
        public Base.ResourceReference? Party { get; set; }
    }

    public class Composition_Event : BackboneElement
    {
        public Base.CodeableConcept[]? Code { get; set; }
        public Base.ResourceReference[]? Detail { get; set; }
        public Base.Period? Period { get; set; }
    }

    public class Composition_RelatesTo : BackboneElement
    {
        public required string Code { get; set; }
        public object? Target    
        {
            get
            {
                if (TargetReference is not null)
                {
                    return TargetReference;
                }
        
                if (TargetIdentifier is not null)
                {
                    return TargetIdentifier;
                }
        
                return null;
            }
        
            set
            {
                if (value?.GetType() == typeof(Base.ResourceReference))
                {
                    TargetReference = (Base.ResourceReference)value;return;
                }
        
                if (value?.GetType() == typeof(Base.Identifier))
                {
                    TargetIdentifier = (Base.Identifier)value;return;
                }
        
                throw new ArgumentException("Invalid type provided");
            }
        }
        public Base.ResourceReference? TargetReference { get; set; }
        public Base.Identifier? TargetIdentifier { get; set; }
    }
}